### ISSUE TITLE
AttributeError due to circular imports

### ISSUE THREAD

**Issue by anmull** [radiac/django-tagulous#124]
Long story short, I have a less common scenario that exposed this issue. Initially the error was never exposed due to the order of imports (don't ask me why, it was weird), but once isort was run I saw the error.

The underlying issue is that `tagulous.forms` imports `tagulous.models.options` which ultimately results in `tagulous.fields` being imported, which imports `tagulous.forms` so it is only partially initialized.

I verified it by moving the `from .models import options` and `from .models.models import BaseTagModel, TagModelQuerySet` statements inside the methods that referenced them and everything works.

Obviously the fast and ugly way is to just put import statements in the methods, but I'm not a fan of that so I just wanted to mention the issue since you know best what can be moved around without a large refactor.

If you need me to verify any changes let me know.

Here's the (edited) traceback

```
  File ".../common/models.py", line 6, in <module>
    from tags.models import TagModelBase
  File ".../tags/models.py", line 2, in <module>
    from .admin import TagAdminBase
  File ".../tags/admin.py", line 3, in <module>
    from tagulous import admin as tag_admin
  File ".../lib/python3.9/site-packages/tagulous/admin.py", line 8, in <module>
    from . import forms as tag_forms
  File ".../lib/python3.9/site-packages/tagulous/forms.py", line 13, in <module>
    from .models import options
  File ".../lib/python3.9/site-packages/tagulous/models/__init__.py", line 6, in <module>
    from ..signals.pre import register_pre_signals
  File ".../lib/python3.9/site-packages/tagulous/signals/pre.py", line 10, in <module>
    from ..models.tagged import TaggedModel
  File ".../lib/python3.9/site-packages/tagulous/models/tagged.py", line 13, in <module>
    from .fields import (
  File ".../lib/python3.9/site-packages/tagulous/models/fields.py", line 308, in <module>
    class SingleTagField(BaseTagField, models.ForeignKey):
  File ".../lib/python3.9/site-packages/tagulous/models/fields.py", line 367, in SingleTagField
    def formfield(self, form_class=forms.SingleTagField, **kwargs):
AttributeError: partially initialized module 'tagulous.forms' has no attribute 'SingleTagField' (most likely due to a circular import)
```

**Comment by radiac** [comment]
@anmull Apologies for the delay looking into this - I can see where this circular import is happening, and want to fix it, but I can't seem to trigger it to see if it's fixed!

It looks like you're triggering it with `from tagulous import admin as tag_admin` in your `models.py`, but when I put that in one of my models it's not triggering. There must be a race condition somewhere which I'm not recreating - I know it was a while back, but can you give any more info about how I can recreate it?

Failing that, I've got a potential fix on a branch - if you'd like to be a guinea pig, you can try it with:

    pip install -e git+https://github.com/radiac/django-tagulous.git@bugfix/circular-import#egg=django-tagulous

**Comment by anmull** [comment]
Apologies for not replying sooner. I just tested this with a new install using the old module and I can confirm that it does remove the error.

Thanks!

**Comment by radiac** [comment]
Brilliant, thanks for testing!

I've merged that change into the main branch and released it under v1.3.0. Closing this as hopefully it's now resolved.

**Comment by anmull** [comment]
> Brilliant, thanks for testing!
> 
> I've merged that change into the main branch and released it under v1.3.0. Closing this as hopefully it's now resolved.

I reinstalled the venv with HEAD last night and it worked perfectly. Thanks for the fix
